src/mg/PAMI/Message/IncomingMessage.php

```diff
@@ -81,6 +87,33 @@ public function getRawContent()
     }
 
     /**
+     * Returns the channel variables for all reported channels.
+     * https://github.com/marcelog/PAMI/issues/85
+     *
+     * The channel names will be lowercased.
+     *
+     * @return array
+     */
+    public function getAllChannelVariables()
+    {
+      return $this->channelVariables;
+    }
+
+    /**
+     * Returns the channel variables for the given channel.
+     * https://github.com/marcelog/PAMI/issues/85
+     *
+     * @param string $channel Channel name. If not given, will return variables
+     * for the "current" channel.
+     *
+     * @return array
+     */
+    public function getChannelVariables($channel = 'default')
```

Comment:
Upon reflection, one way to fix that may be to defer parsing these variables to the end of this method, so that the current channel name is known.

Then parsing these messages and comparing the channel name to see if they're for the default channel would work.

Also, I would update the `::getChannelVariables()` method as follows:

```php
public function getChannelVariables($channel = null)
{
    if ($channel === null) {
        $channel = $this->getChannel();
    }

    return $this->channelVariables[strtolower($channel)];
}
```